src/ray/rpc/server_call.h

```diff
@@ -221,24 +244,32 @@ class ServerCallImpl : public ServerCall {
       ray::stats::STATS_grpc_server_req_handling.Record(1.0, call_name_);
     }
     if (!io_service_.stopped()) {
-      io_service_.post([this, auth_success] { HandleRequestImpl(auth_success); },
-                       call_name_ + ".HandleRequestImpl",
-                       // Implement the delay of the rpc server call as the
-                       // delay of HandleRequestImpl().
-                       ray::asio::testing::GetDelayUs(call_name_));
+      io_service_.post(
+          [this, auth_success, token_auth_failed, cluster_id_auth_failed] {
+            HandleRequestImpl(auth_success, token_auth_failed, cluster_id_auth_failed);
+          },
+          call_name_ + ".HandleRequestImpl",
+          // Implement the delay of the rpc server call as the
+          // delay of HandleRequestImpl().
+          ray::asio::testing::GetDelayUs(call_name_));
     } else {
       // Handle service for rpc call has stopped, we must handle the call here
       // to send reply and remove it from cq
       RAY_LOG(DEBUG) << "Handle service has been closed.";
       if (auth_success) {
         SendReply(Status::Invalid("HandleServiceClosed"));
+      } else if (token_auth_failed) {
+        SendReply(Status::AuthError(
+            "InvalidAuthToken: Authentication token is missing or incorrect"));
```

Comment:
We should ensure that we're using gRPC's UNAUTHENTICATED status code for these errors: https://grpc.github.io/grpc/core/md_doc_statuscodes.html

Reply:
let's also rename `Status::AuthError` to `Status::Unauthenticated` to match